sopel/modules/safety.py

```diff
@@ -7,75 +7,112 @@
 
-
 import requests
 
 from sopel import formatting, plugin, tools
-from sopel.config import types
+from sopel.bot import Sopel
+from sopel.config import Config, types
+from sopel.trigger import Trigger
 
 
 LOGGER = logging.getLogger(__name__)
 PLUGIN_OUTPUT_PREFIX = '[safety] '
 
-vt_base_api_url = 'https://www.virustotal.com/vtapi/v2/url/'
-malware_domains = set()
+SAFETY_MODES = ["off", "local", "local strict", "on", "strict"]
+VT_API_URL = "https://www.virustotal.com/api/v3/urls"
+CACHE_LIMIT = 512
 known_good = []
-cache_limit = 512
 
 
 class SafetySection(types.StaticSection):
-    enabled_by_default = types.BooleanAttribute('enabled_by_default', default=True)
-    """Whether to enable URL safety in all channels where it isn't explicitly disabled."""
+    enabled_by_default = types.BooleanAttribute("enabled_by_default", default=True)
+    """Deprecated: Sets default_mode to "off" or "on"."""
+    default_mode = types.ValidatedAttribute("default_mode")
+    """Which mode to use in channels without a mode set."""
     known_good = types.ListAttribute('known_good')
     """List of "known good" domains to ignore."""
     vt_api_key = types.ValidatedAttribute('vt_api_key')
     """Optional VirusTotal API key (improves malicious URL detection)."""
+    domain_blocklist_url = types.ValidatedAttribute("domain_blocklist_url")
+    """Optional hosts-file formatted domain blocklist to use instead of StevenBlack's."""
 
 
-def configure(config):
+def configure(settings: Config):
     """
     | name | example | purpose |
     | ---- | ------- | ------- |
-    | enabled\\_by\\_default | True | Enable URL safety in all channels where it isn't explicitly disabled. |
-    | known\\_good | sopel.chat,dftba.net | List of "known good" domains to ignore. |
+    | default\\_mode | on | Which mode to use in channels without a mode set. |
+    | known\\_good | sopel.chat,dftba.net | List of "known good" domains or regexes to ignore. This can save VT API calls. |
     | vt\\_api\\_key | 0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef | Optional VirusTotal API key to improve malicious URL detection |
+    | domain\\_blocklist\\_url | https://example.com/bad-hosts.txt | Optional hosts-file formatted domain blocklist to use instead of StevenBlack's. |
     """
-    config.define_section('safety', SafetySection)
-    config.safety.configure_setting(
-        'enabled_by_default',
-        "Enable URL safety in channels that don't specifically disable it?",
+    settings.define_section("safety", SafetySection)
+    settings.safety.configure_setting(
+        "default_mode",
+        (
+            "Which mode should be used in channels that haven't specifically set one?"
+            "\n({})".format("/".join(SAFETY_MODES))
+        ),
+        default="on",
     )
-    config.safety.configure_setting(
+    settings.safety.configure_setting(
         'known_good',
-        'Enter any domains to whitelist',
+        "Enter any domains to allowlist",
```

Comment:
Using "allowlist" as a verb is stylistically icky. Using it as a single word in any sense is icky, too, for that matter.

I think changes like this come across as performative, and [ignore the original derivation](https://seattlecollegian.com/op-ed-blacklist-and-whitelist-arent-racist-words-you-are/) of the terms, but that's not a good reason to block this whole PR. I'm not that petty. Just… If you're dead set on eliminating the word "whitelist" let's please at least change it to something that sounds like English. Some alternatives:

```suggestion
        "Enter any domains to trust",
```

```suggestion
        "Enter any domains to consider trusted",
```

Reply:
I'd changed the last bit in one place, but not in the other two. Why are plugin config options documented in three separate places? I understand doc phrasing vs prompt phrasing, but where's the table used?

Reply:
The table gets slurped up by the part of the website that generates [plugin configuration info](https://sopel.chat/usage/plugin-configuration/). Is there a better way? Absolutely there is, but no one has bothered refactoring this bit yet. 🤷‍♂️